exercises/ex05-reminder/reminder/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn event_counter)]
+	pub type EventCounter<T> = StorageValue<_, u32, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::unbounded]
+	#[pallet::getter(fn reminders)]
+	pub type Reminders<T: Config> =
+		StorageMap<_, Blake2_256, T::BlockNumber, Vec<Vec<u8>>, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		ReminderSet(T::BlockNumber, Vec<u8>),
+		Reminder(Vec<u8>),
+		RemindersExecuteds(u32),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(n: T::BlockNumber) -> Weight {
+			let reminders = Self::reminders(n);
+			for reminder in reminders {
+				Self::deposit_event(Event::Reminder(reminder.clone()));
+				<EventCounter<T>>::mutate(|value| *value += 1);
+			}
+			<Reminders<T>>::remove(n);
+			0
```

Comment:
To respect the way hooks are supposed to be used, we should return the actual weight used, not 0.
Do you think you can include this?

Reply:
Yes, but it will be harder to test. I'll find a way